geoserver/src/eumetsat/pinning/src/main/java/org/geoserver/eumetsat/pinning/PinningService.java

```diff
@@ -0,0 +1,355 @@
+        viewsEvaluator.fullPinReset();
+        logger.log(Level.INFO, "Inserting views");
+
+        for (View remoteView : remoteViews) {
+            ParsedView view = viewsClient.parseView(remoteView);
+            if (isLiveView(view)) {
+                continue;
+            }
+
+            ViewRecord viewRecord = viewsEvaluator.buildView(view);
+            viewsEvaluator.addViewAndPin(viewRecord);
+        }
+        viewsEvaluator.flushBatches();
+        viewsEvaluator.updateLastUpdate(lastUpdate);
+    }
+
+    /**
+     * Retrieves the current status of the pinning process.
+     *
+     * @return A PinningStatus object containing the current task ID and status
+     */
+    public PinningStatus getStatus() {
+        UUID currentId = currentTaskId.get();
+        String status = taskStatus.get();
+        return new PinningStatus(
+                currentId != null ? currentId.toString() : "No taskID available since last restart",
+                status);
+    }
+
+    /**
+     * Attempts to acquire an advisory lock for the pinning process.
+     *
+     * @param conn The database connection to use for acquiring the lock
+     * @return true if the lock is successfully acquired, false otherwise
+     * @throws SQLException if a database access error occurs
+     */
+    private boolean acquireLock(Connection conn) throws SQLException {
+        try (PreparedStatement stmt = conn.prepareStatement("SELECT pg_try_advisory_lock(?)")) {
+            stmt.setInt(1, PINNING_LOCK_ID);
+            logger.log(Level.FINEST, "Acquiring Advisory Lock");
+            ResultSet rs = stmt.executeQuery();
+            if (rs.next() && rs.getBoolean(1)) {
+                logger.log(Level.FINEST, "Advisory Lock acquired");
+                return true;
+            } else {
+                logger.log(Level.FINE, "Unable to acquire the Lock. Is the lock already in use?");
+                UUID taskId = currentTaskId.get();
+                logger.log(Level.WARNING, "Another process is holding the lock: " + taskId);
+                return false;
+            }
+        }
+    }
+
+    /**
+     * Releases the advisory lock on the database connection.
+     *
+     * @param conn The database connection used to release the lock
+     */
+    private void releaseLock(Connection conn) {
+        try (PreparedStatement stmt = conn.prepareStatement("SELECT pg_advisory_unlock(?)")) {
+            logger.log(Level.FINEST, "Releasing Advisory Lock");
+            stmt.setInt(1, PINNING_LOCK_ID);
+            stmt.execute();
+            logger.log(Level.FINEST, "Advisory Lock released");
+        } catch (Exception e) {
+            logger.log(
+                    Level.SEVERE,
+                    "Unable to release the Advisory Lock due to " + e.getLocalizedMessage());
+        }
+    }
+
+    private boolean isLiveView(ParsedView parsed) {
+        boolean isLiveView = !"absolute".equalsIgnoreCase(parsed.getTimeMode());
```

Comment:
The two modes are absolute and latest, there is nothing else? Reminder we don't control the user view handling, a different group works on that.
I'd do the right thing with the modes we are aware of, and have an warning/severe log  for modes not recognized? "Skipping view xyz, mode 'Supercalifragilisticexpialidocious' was not recognized".